### ISSUE TITLE
Unable to install Tentacle when Space contains a whitespace character

### ISSUE THREAD

**Issue by rchildress87** [OctopusDeploy/OctopusDSC#303]
Summary:
When installing an Octopus Tentacle agent on a Windows computer with the OctopusDSC module, the DSC process hangs indefinitely and the installation never completes:
```powershell
PS > Start-DscConfiguration -Path .\VMConfig\ -Wait -Verbose

...

VERBOSE: [dsc-test]: LCM:  [ Start  Resource ]  [[cTentacleAgent]OctopusTentacleSpace::[OctopusTentacle]Cloud Infrastructure]
VERBOSE: [dsc-test]: LCM:  [ Start  Test     ]  [[cTentacleAgent]OctopusTentacleSpace::[OctopusTentacle]Cloud Infrastructure]
VERBOSE: [dsc-test]:                            [[cTentacleAgent]OctopusTentacleSpace::[OctopusTentacle]Cloud Infrastructure] Checking if Tentacle is installed
VERBOSE: [dsc-test]:                            [[cTentacleAgent]OctopusTentacleSpace::[OctopusTentacle]Cloud Infrastructure] Tentacle present: False
VERBOSE: [dsc-test]:                            [[cTentacleAgent]OctopusTentacleSpace::[OctopusTentacle]Cloud Infrastructure] Checking for Windows Service: OctopusDeploy Tentacle: Tentacle-dev-Cloud Infrastructure
VERBOSE: [dsc-test]:                            [[cTentacleAgent]OctopusTentacleSpace::[OctopusTentacle]Cloud Infrastructure] Windows service: Not installed
VERBOSE: [dsc-test]:                            [[cTentacleAgent]OctopusTentacleSpace::[OctopusTentacle]Cloud Infrastructure] Ensure: Absent vs. Present = False
VERBOSE: [dsc-test]: LCM:  [ End    Test     ]  [[cTentacleAgent]OctopusTentacleSpace::[OctopusTentacle]Cloud Infrastructure]  in 0.1250 seconds.
VERBOSE: [dsc-test]: LCM:  [ Start  Set      ]  [[cTentacleAgent]OctopusTentacleSpace::[OctopusTentacle]Cloud Infrastructure]
VERBOSE: [dsc-test]:                            [[cTentacleAgent]OctopusTentacleSpace::[OctopusTentacle]Cloud Infrastructure] Checking if Tentacle is installed
VERBOSE: [dsc-test]:                            [[cTentacleAgent]OctopusTentacleSpace::[OctopusTentacle]Cloud Infrastructure] Tentacle present: False
VERBOSE: [dsc-test]:                            [[cTentacleAgent]OctopusTentacleSpace::[OctopusTentacle]Cloud Infrastructure] Checking for Windows Service: OctopusDeploy Tentacle: Tentacle-dev-Cloud Infrastructure
VERBOSE: [dsc-test]:                            [[cTentacleAgent]OctopusTentacleSpace::[OctopusTentacle]Cloud Infrastructure] Windows service: Not installed
VERBOSE: [dsc-test]:                            [[cTentacleAgent]OctopusTentacleSpace::[OctopusTentacle]Cloud Infrastructure] Configuring Tentacle...
VERBOSE: [dsc-test]:                            [[cTentacleAgent]OctopusTentacleSpace::[OctopusTentacle]Cloud Infrastructure] Installing Tentacle...
VERBOSE: [dsc-test]:                            [[cTentacleAgent]OctopusTentacleSpace::[OctopusTentacle]Cloud Infrastructure] Beginning Tentacle installation
VERBOSE: [dsc-test]:                            [[cTentacleAgent]OctopusTentacleSpace::[OctopusTentacle]Cloud Infrastructure] Tentacle Home directory does not exist. Creating...
VERBOSE: [dsc-test]:                            [[cTentacleAgent]OctopusTentacleSpace::[OctopusTentacle]Cloud Infrastructure] Downloading Octopus Tentacle MSI from https://download.octopusdeploy.com/octopus/Octopus.Tentacle.6.2.277-x64.msi to C:\Octopus\Cloud Infrastructure\Tentacle.msi
VERBOSE: [dsc-test]:                            [[cTentacleAgent]OctopusTentacleSpace::[OctopusTentacle]Cloud Infrastructure] Checking to see if we have an installer at C:\Octopus\Cloud Infrastructure\Tentacle.msi
VERBOSE: [dsc-test]:                            [[cTentacleAgent]OctopusTentacleSpace::[OctopusTentacle]Cloud Infrastructure] No local installer found
VERBOSE: [dsc-test]:                            [[cTentacleAgent]OctopusTentacleSpace::[OctopusTentacle]Cloud Infrastructure] Downloading https://download.octopusdeploy.com/octopus/Octopus.Tentacle.6.2.277-x64.msi to C:\Octopus\Cloud Infrastructure\Tentacle.msi
VERBOSE: [dsc-test]:                            [[cTentacleAgent]OctopusTentacleSpace::[OctopusTentacle]Cloud Infrastructure] Configured temp folder does not currently exist, creating...
VERBOSE: [dsc-test]:                            [[cTentacleAgent]OctopusTentacleSpace::[OctopusTentacle]Cloud Infrastructure] Installing MSI...
VERBOSE: [dsc-test]:                            [[cTentacleAgent]OctopusTentacleSpace::[OctopusTentacle]Cloud Infrastructure] Executing 'msiexec.exe /i C:\Octopus\Cloud Infrastructure\Tentacle.msi /quiet /l*v C:\Octopus\logs\Tentacle.Tentacle-dev-Cloud Infrastructure.msi.log'
```
This is because the path specified for msiexec logging includes a whitespace character and is not wrapped in quotes. Additionally, not log file is ever produced.

Expectation:
The msiexec installation of Octopus Tentacle agent completes successfully and a log file is created in the provided path.

Workaround:
Provide the cTentacleAgent DSC resource with space name(s) that do not contain whitespace character(s).

**Comment by rchildress87** [comment]
The PR associated with this issue was tested by modifying the module locally:
```powershell
PS > Start-DscConfiguration -Path .\VMConfig\ -Wait -Verbose

...

VERBOSE: [dsc-test]: LCM:  [ Start  Resource ]  [[cTentacleAgent]OctopusTentacleSpace::[OctopusTentacle]Cloud Infrastructure]
VERBOSE: [dsc-test]: LCM:  [ Start  Test     ]  [[cTentacleAgent]OctopusTentacleSpace::[OctopusTentacle]Cloud Infrastructure]
VERBOSE: [dsc-test]:                            [[cTentacleAgent]OctopusTentacleSpace::[OctopusTentacle]Cloud Infrastructure] Checking if Tentacle is installed
VERBOSE: [dsc-test]:                            [[cTentacleAgent]OctopusTentacleSpace::[OctopusTentacle]Cloud Infrastructure] Tentacle present: False
VERBOSE: [dsc-test]:                            [[cTentacleAgent]OctopusTentacleSpace::[OctopusTentacle]Cloud Infrastructure] Checking for Windows Service: OctopusDeploy Tentacle: Tentacle-dev-Cloud Infrastructure
VERBOSE: [dsc-test]:                            [[cTentacleAgent]OctopusTentacleSpace::[OctopusTentacle]Cloud Infrastructure] Windows service: Not installed
VERBOSE: [dsc-test]:                            [[cTentacleAgent]OctopusTentacleSpace::[OctopusTentacle]Cloud Infrastructure] Ensure: Absent vs. Present = False
VERBOSE: [dsc-test]: LCM:  [ End    Test     ]  [[cTentacleAgent]OctopusTentacleSpace::[OctopusTentacle]Cloud Infrastructure]  in 0.2500 seconds.
VERBOSE: [dsc-test]: LCM:  [ Start  Set      ]  [[cTentacleAgent]OctopusTentacleSpace::[OctopusTentacle]Cloud Infrastructure]
VERBOSE: [dsc-test]:                            [[cTentacleAgent]OctopusTentacleSpace::[OctopusTentacle]Cloud Infrastructure] Checking if Tentacle is installed
VERBOSE: [dsc-test]:                            [[cTentacleAgent]OctopusTentacleSpace::[OctopusTentacle]Cloud Infrastructure] Tentacle present: False
VERBOSE: [dsc-test]:                            [[cTentacleAgent]OctopusTentacleSpace::[OctopusTentacle]Cloud Infrastructure] Checking for Windows Service: OctopusDeploy Tentacle: Tentacle-dev-Cloud Infrastructure
VERBOSE: [dsc-test]:                            [[cTentacleAgent]OctopusTentacleSpace::[OctopusTentacle]Cloud Infrastructure] Windows service: Not installed
VERBOSE: [dsc-test]:                            [[cTentacleAgent]OctopusTentacleSpace::[OctopusTentacle]Cloud Infrastructure] Configuring Tentacle...
VERBOSE: [dsc-test]:                            [[cTentacleAgent]OctopusTentacleSpace::[OctopusTentacle]Cloud Infrastructure] Installing Tentacle...
VERBOSE: [dsc-test]:                            [[cTentacleAgent]OctopusTentacleSpace::[OctopusTentacle]Cloud Infrastructure] Beginning Tentacle installation
VERBOSE: [dsc-test]:                            [[cTentacleAgent]OctopusTentacleSpace::[OctopusTentacle]Cloud Infrastructure] Tentacle Home directory does not exist. Creating...
VERBOSE: [dsc-test]:                            [[cTentacleAgent]OctopusTentacleSpace::[OctopusTentacle]Cloud Infrastructure] Downloading Octopus Tentacle MSI from https://download.octopusdeploy.com/octopus/Octopus.Tentacle.6.2.277-x64.msi to C:\Octopus\Cloud Infrastructure\Tentacle.msi
VERBOSE: [dsc-test]:                            [[cTentacleAgent]OctopusTentacleSpace::[OctopusTentacle]Cloud Infrastructure] Checking to see if we have an installer at C:\Octopus\Cloud Infrastructure\Tentacle.msi
VERBOSE: [dsc-test]:                            [[cTentacleAgent]OctopusTentacleSpace::[OctopusTentacle]Cloud Infrastructure] No local installer found
VERBOSE: [dsc-test]:                            [[cTentacleAgent]OctopusTentacleSpace::[OctopusTentacle]Cloud Infrastructure] Downloading https://download.octopusdeploy.com/octopus/Octopus.Tentacle.6.2.277-x64.msi to C:\Octopus\Cloud Infrastructure\Tentacle.msi
VERBOSE: [dsc-test]:                            [[cTentacleAgent]OctopusTentacleSpace::[OctopusTentacle]Cloud Infrastructure] Installing MSI...
VERBOSE: [dsc-test]:                            [[cTentacleAgent]OctopusTentacleSpace::[OctopusTentacle]Cloud Infrastructure] Executing 'msiexec.exe /i C:\Octopus\Cloud Infrastructure\Tentacle.msi /quiet /l*v C:\Octopus\logs\Tentacle.Tentacle-dev-Cloud Infrastructure.msi.log'
VERBOSE: [dsc-test]:                            [[cTentacleAgent]OctopusTentacleSpace::[OctopusTentacle]Cloud Infrastructure] MSI installer returned exit code 0
VERBOSE: [dsc-test]:                            [[cTentacleAgent]OctopusTentacleSpace::[OctopusTentacle]Cloud Infrastructure] Configuring and registering Tentacle
VERBOSE: [dsc-test]:                            [[cTentacleAgent]OctopusTentacleSpace::[OctopusTentacle]Cloud Infrastructure] Tentacle configuration set as C:\Octopus\Cloud Infrastructure\Tentacle-dev-Cloud Infrastructure\Tentacle.config
VERBOSE: [dsc-test]:                            [[cTentacleAgent]OctopusTentacleSpace::[OctopusTentacle]Cloud Infrastructure] Executing command 'C:\Program Files\Octopus Deploy\Tentacle\\Tentacle.exe create-instance --instance Tentacle-dev-Cloud Infrastructure --config C:\Octopus\Cloud Infrastructure\Tentacle-dev-Cloud
Infrastructure\Tentacle.config --console'
VERBOSE: [dsc-test]:                            [[cTentacleAgent]OctopusTentacleSpace::[OctopusTentacle]Cloud Infrastructure] Creating empty configuration file: C:\Octopus\Cloud Infrastructure\Tentacle-dev-Cloud Infrastructure\Tentacle.config Setting home directory to: C:\Octopus\Cloud Infrastructure\Tentacle-dev-Cloud Infrastructure Saving
instance: Tentacle-dev-Cloud Infrastructure
VERBOSE: [dsc-test]:                            [[cTentacleAgent]OctopusTentacleSpace::[OctopusTentacle]Cloud Infrastructure] done.
VERBOSE: [dsc-test]:                            [[cTentacleAgent]OctopusTentacleSpace::[OctopusTentacle]Cloud Infrastructure] Executing command 'C:\Program Files\Octopus Deploy\Tentacle\\Tentacle.exe new-certificate --instance Tentacle-dev-Cloud Infrastructure --console'
VERBOSE: [dsc-test]:                            [[cTentacleAgent]OctopusTentacleSpace::[OctopusTentacle]Cloud Infrastructure] A new certificate has been generated and installed. Thumbprint: 5C0EDC08AD62F121880A66DE0CAC87F8B500EBE3 These changes require a restart of the Tentacle.
VERBOSE: [dsc-test]:                            [[cTentacleAgent]OctopusTentacleSpace::[OctopusTentacle]Cloud Infrastructure] done.
VERBOSE: [dsc-test]:                            [[cTentacleAgent]OctopusTentacleSpace::[OctopusTentacle]Cloud Infrastructure] Executing command 'C:\Program Files\Octopus Deploy\Tentacle\\Tentacle.exe configure --instance Tentacle-dev-Cloud Infrastructure --home C:\Octopus\Cloud Infrastructure --app C:\Octopus\Cloud Infrastructure\Applications
--console --noListen True'
VERBOSE: [dsc-test]:                            [[cTentacleAgent]OctopusTentacleSpace::[OctopusTentacle]Cloud Infrastructure] Home directory set to: C:\Octopus\Cloud Infrastructure Application directory set to: C:\Octopus\Cloud Infrastructure\Applications Tentacle will not listen on a port These changes require a restart of the Tentacle.
VERBOSE: [dsc-test]:                            [[cTentacleAgent]OctopusTentacleSpace::[OctopusTentacle]Cloud Infrastructure] done.
VERBOSE: [dsc-test]:                            [[cTentacleAgent]OctopusTentacleSpace::[OctopusTentacle]Cloud Infrastructure] Executing command 'C:\Program Files\Octopus Deploy\Tentacle\\Tentacle.exe service --install --instance Tentacle-dev-Cloud Infrastructure --console --reconfigure'
VERBOSE: [dsc-test]:                            [[cTentacleAgent]OctopusTentacleSpace::[OctopusTentacle]Cloud Infrastructure] Service installed Service reconfigured
VERBOSE: [dsc-test]:                            [[cTentacleAgent]OctopusTentacleSpace::[OctopusTentacle]Cloud Infrastructure] done.
VERBOSE: [dsc-test]:                            [[cTentacleAgent]OctopusTentacleSpace::[OctopusTentacle]Cloud Infrastructure] Adding Tentacle to worker pool
VERBOSE: [dsc-test]:                            [[cTentacleAgent]OctopusTentacleSpace::[OctopusTentacle]Cloud Infrastructure] Adding dsc-test to pool(s) dev
VERBOSE: [dsc-test]:                            [[cTentacleAgent]OctopusTentacleSpace::[OctopusTentacle]Cloud Infrastructure] Executing command 'C:\Program Files\Octopus Deploy\Tentacle\\Tentacle.exe register-worker --instance Tentacle-dev-Cloud Infrastructure --server https://[redacted].octopus.app/ --name Octopus-Vm-dev --force --space Cloud
 Infrastructure --apiKey ********** --comms-style TentacleActive --server-comms-port 10943 --workerpool dev'
VERBOSE: [dsc-test]:                            [[cTentacleAgent]OctopusTentacleSpace::[OctopusTentacle]Cloud Infrastructure] Checking connectivity on the server communications port 10943... Connected successfully Registering the tentacle with the server at https://[redacted].octopus.app/ Detected automation environment: NoneOrUnknown Machine
registered successfully These changes require a restart of the Tentacle.
VERBOSE: [dsc-test]:                            [[cTentacleAgent]OctopusTentacleSpace::[OctopusTentacle]Cloud Infrastructure] done.
VERBOSE: [dsc-test]:                            [[cTentacleAgent]OctopusTentacleSpace::[OctopusTentacle]Cloud Infrastructure] Tentacle commands complete
VERBOSE: [dsc-test]:                            [[cTentacleAgent]OctopusTentacleSpace::[OctopusTentacle]Cloud Infrastructure] Tentacle installed!
VERBOSE: [dsc-test]:                            [[cTentacleAgent]OctopusTentacleSpace::[OctopusTentacle]Cloud Infrastructure] Starting OctopusDeploy Tentacle: Tentacle-dev-Cloud Infrastructure
VERBOSE: [dsc-test]:                            [[cTentacleAgent]OctopusTentacleSpace::[OctopusTentacle]Cloud Infrastructure] Finished
VERBOSE: [dsc-test]: LCM:  [ End    Set      ]  [[cTentacleAgent]OctopusTentacleSpace::[OctopusTentacle]Cloud Infrastructure]  in 20.0640 seconds.
VERBOSE: [dsc-test]: LCM:  [ End    Resource ]  [[cTentacleAgent]OctopusTentacleSpace::[OctopusTentacle]Cloud Infrastructure]
```

**Comment by benPearce1** [comment]
Resolved by #304
